Refactoring and cleaning

jubianchi opened the pull request:
I tried to refactor and clean the code as much as I could.

@pierrejoye I sent a PR because I don't know if you have work in progress which could conflict with my changes. Also, perhaps you are not fine with the changes I did ;)
If you want me to change anything or revert some part of the patch, feel free to ask.

jubianchi committed:
Clean HTML

jubianchi committed:
Add 404 page

jubianchi committed:
Removed exit and header calls

jubianchi committed:
Rearrange templates directory

jubianchi committed:
Persist users' json data

jubianchi committed:
Refactored application

jubianchi committed:
Fix a condition

pierrejoye commented:
No work on progress that I cannot easily migrate, go ahead with the cleanup
of my mess! :)
On Apr 10, 2015 6:47 PM, "Julien BIANCHI" notifications@github.com wrote:

> I tried to refactor and clean the code as much as I could.
> 
> @pierrejoye https://github.com/pierrejoye I sent a PR because I don't
> know if you have work in progress which could conflict with my changes.
> Also, perhaps you are not fine with the changes I did ;)
> If you want me to change naything or revert some part of the patch, feel
> 
> ## free to ask.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/FriendsOfPHP/pickleweb/pull/10
> Commit Summary
> - Clean HTML
> - Add 404 page
> - Removed exit and header calls
> - Rearrange templates directory
> - Persist users' json data
> - Refactored application
> 
> File Changes
> - _M_ .gitignore
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-0 (1)
> - _M_ src/Action/AuthAction.php
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-1 (19)
> - _A_ src/Application.php
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-2 (84)
> - _M_ src/Auth/GithubProvider.php
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-3 (25)
> - _M_ src/View/Twig.php
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-4 (18)
> - _M_ templates/account.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-5 (27)
> - _A_ templates/errors/404.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-6 (5)
> - _A_ templates/errors/500.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-7 (5)
> - _A_ templates/errors/generic.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-8 (11)
> - _M_ templates/extension_register_info.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-9 (6)
> - _M_ templates/home.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-10 (10)
> - _M_ templates/index.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-11 (19)
> - _M_ templates/package.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-12 (4)
> - _A_ templates/partials/header.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-13 (16)
> - _R_ templates/partials/search.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-14 (0)
> - _M_ templates/register.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-15 (2)
> - _M_ templates/registerextension.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-16 (6)
> - _D_ templates/registergithub.html
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-17 (6)
> - _M_ web/index.php
>   https://github.com/FriendsOfPHP/pickleweb/pull/10/files#diff-18
>   (236)
> 
> Patch Links:
> - https://github.com/FriendsOfPHP/pickleweb/pull/10.patch
> - https://github.com/FriendsOfPHP/pickleweb/pull/10.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/FriendsOfPHP/pickleweb/pull/10.

pierrejoye commented:
Btw, we may use the middleware layer for the auth, could be cleaner :)

jubianchi commented:
@pierrejoye you are right ;) I did not had enough time to read the Slim doc so I kept the actual logic but yeah, I would be great to use those middlewares.

jubianchi committed:
Enable twig cache only for production

jubianchi committed:
Implement extension definition storage

Here is an example JSON definition:

```json
{
    "packages": {
        "xdebug": {
            "dev-master": {
                "description": "The Xdebug extension helps you debugging
your script by providing a lot of valuable debug information",
                "homepage": "http://xdebug.org/",
                "time": "2015-04-10",
                "license": "The Xdebug License, version 1.01",
                "authors": [
                    {
                        "name": "Derick Rethans",
                        "email": "derick@php.net",
                        "homepage": "http://derickrethans.nl",
                        "role": "Author"
                    }
                ],
                "support": {
                    "issues": "http://bugs.xdebug.org/",
                    "email": "xdebug-general@lists.xdebug.org",
                    "source": "https://github.com/xdebug/xdebug"
                }
            },

            "2.3.2": {
                "description": "The Xdebug extension helps you debugging
your script by providing a lot of valuable debug information",
                "homepage": "http://xdebug.org/",
                "time": "2015-03-22",
                "license": "The Xdebug License, version 1.01",
                "authors": [
                    {
                        "name": "Derick Rethans",
                        "email": "derick@php.net",
                        "homepage": "http://derickrethans.nl",
                        "role": "Author"
                    }
                ],
                "support": {
                    "issues": "http://bugs.xdebug.org/",
                    "email": "xdebug-general@lists.xdebug.org",
                    "source": "https://github.com/xdebug/xdebug"
                }
            },

            "2.3.1": {
                "description": "The Xdebug extension helps you debugging
your script by providing a lot of valuable debug information",
                "homepage": "http://xdebug.org/",
                "time": "2015-02-24",
                "license": "The Xdebug License, version 1.01",
                "authors": [
                    {
                        "name": "Derick Rethans",
                        "email": "derick@php.net",
                        "homepage": "http://derickrethans.nl",
                        "role": "Author"
                    }
                ],
                "support": {
                    "issues": "http://bugs.xdebug.org/",
                    "email": "xdebug-general@lists.xdebug.org",
                    "source": "https://github.com/xdebug/xdebug"
                }
            }
        }
    }
}

```
